src/auth/roles/index.ts

```diff
@@ -1,146 +1,137 @@
-      return result.rows[0];
-    },
-    getUserRole: async ({ userId, organisationId }) => {
-      const query = await client('user_account_access_role')
-        .select('access_role_id')
-        .where({ user_account_id: userId, organisation_id: organisationId });
-      if (query.length === 0) {
-        throw new Error('User does not exist');
+    }
+  },
+
+  remove: async (client, { role, userId, organisationId }) => {
+    const deleteRow = await client('user_account_access_role')
+      .where({
+        user_account_id: userId,
+        organisation_id: organisationId,
+        access_role_id: client.select('access_role_id')
+          .table('access_role')
+          .where({ ['access_role_name']: role }),
+      })
+      .del()
+      .returning('*');
+
+    if (deleteRow.length === 0) {
+      throw new Error(
+        `User ${userId} is not associated with role ${role} at organisation ${organisationId}`
+      );
+    }
+    return deleteRow[0];
+  },
+
+  move: async (client, { to, from , userId, organisationId }) => {
+    const userHasSourceRole = await Roles.userHas(client, { role: from, userId, organisationId });
+
+    if (! userHasSourceRole) {
+      throw new Error(`"from" role ${from} is not associated with user ${userId}`);
+    }
+
+    try {
+      return await client.transaction((trx) =>
+        Roles.remove(trx, { role: from, userId, organisationId })
+          .then(() => Roles.add(trx, { role: to, userId, organisationId }))
+          .then(trx.commit)
+          .catch(trx.rollback)
+      );
+    } catch (error) {
+      switch (error.code) {
+        case '23505':
+          throw new Error(
+            `User ${userId} is already associated with ` +
+            `role ${from} at organistion ${organisationId}`
+          );
+
+        /* istanbul ignore next */
+        default:
+          throw error;
       }
-      return query[0];
-    },
-  };
+    }
+  },
+
+  removeUserFromAll: async (client, { userId, organisationId }) => {
+    const deleteRow = await client('user_account_access_role')
+      .where({
+        user_account_id: userId,
+        organisation_id: organisationId,
+      })
+      .del()
+      .returning('*');
+
+    if (deleteRow.length === 0) {
+      throw new Error(
+        `User ${userId} is not associated to any roles at organisation ${organisationId}`
+      );
+    }
+    return deleteRow;
+  },
+
+  userHas: async (client, { role, userId, organisationId }) => {
+    const inner = client('user_account_access_role')
+      .select()
+      .where({
+        user_account_id: userId,
+        organisation_id: organisationId,
+        access_role_id: client('access_role')
+          .select('access_role_id')
+          .where({ access_role_name: role }),
+      });
+    const { rows } = await client.raw('SELECT EXISTS ?', [inner]);
+    return rows[0].exists;
+  },
+
+  fromUser: async (client, { userId, organisationId }) => {
```

Comment:
@eliascodes can this be changed to `oneFromUser`

Reply:
Oops, sorry, forgot. Will do.